### PR TITLE
Agreements can be deleted

### PR DESCRIPTION
**Pull request by fareed-mohammed**
https://eaflood.atlassian.net/browse/SFI-298

### CODE REVIEW

File: app/routes/models/delete.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        value: false,
+        text: 'No'
+      }
+    ]
+  }
+
+  // If error is passed to model then this error property is added to the model
+  if (error) {
+    this.model.errorMessage = {
+      text: 'Please select if you would like to delete your application.'
```
Comment (steve-dickinson) [inline app/routes/models/delete.js:28]: Maybe change to:

'Please select if you would like to delete your agreement.'

File: app/views/delete.njk [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          id: "back"
+        }
+      }) 
+    }}
+{% endblock %}
+
+{% block content %}
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+    <form method="POST" action="/delete" novalidate>
```
Comment (steve-dickinson) [inline app/views/delete.njk:21]: Use DELETE verb with suggested route

File: app/routes/delete.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  path: '/delete',
+  options: {
+    handler: async (request, h) => {
+      return h.view('delete', new ViewModel())
+    }
+  }
+},
+{
+  method: 'POST',
+  path: '/delete',
```
Comment (steve-dickinson) [inline app/routes/delete.js:17]: Maybe better if the path is

agreement/delete

File: app/plugins/router.js [hunk 4]
```diff
@@ -34,7 +34,8 @@ const routes = [].concat(
   require('../routes/withdraw'),
   require('../routes/withdrawn'),
   require('../routes/agreement'),
-  require('../routes/clear-session')
+  require('../routes/clear-session'),
+  require('../routes/delete')
```
Comment (steve-dickinson) [inline app/plugins/router.js:38]: Maybe better to move into agreement folder in the routes

**Review by steve-dickinson — CHANGES_REQUESTED** [review]
I've left a couple of suggested changes.

File: app/views/delete.njk (hunk 2 above)
Comment (fareed-mohammed) [reply app/views/delete.njk:21]: Keep as POST didn't route properly to DELETE when I changed it.

File: app/routes/models/delete.js (hunk 1 above)
Comment (fareed-mohammed) [reply app/routes/models/delete.js:28]: Done

File: app/routes/models/delete.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+function ViewModel (value, error) {
+  // Constructor function to create logic dependent nunjucks page
+  this.model = {
+    id: 'delete',
+    name: 'delete',
+    fieldset: {
+      legend: {
+        text: 'Are you sure you want to delete your application?',
```
[the remark this answers is not in the document]
Comment (fareed-mohammed) [reply app/routes/models/delete.js:8]: Done

File: app/routes/delete.js (hunk 3 above)
Comment (fareed-mohammed) [reply app/routes/delete.js:17]: Done

[the remark this answers is not in the document]
Comment (fareed-mohammed) [reply app/routes/delete.js:17]: Keep as POST didn't route properly to DELETE when I changed it.

File: app/routes/delete.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
+const joi = require('joi')
+const ViewModel = require('./models/delete')
+const { deleteAgreement } = require('../agreement')
+const cache = require('../cache')
+
+module.exports = [{
+  method: 'GET',
+  path: '/delete',
```
[the remark this answers is not in the document]
Comment (fareed-mohammed) [reply app/routes/delete.js:8]: Done

File: app/plugins/router.js (hunk 4 above)
Comment (fareed-mohammed) [reply app/plugins/router.js:38]: Done

File: app/routes/funding-options/actions.js [hunk 7]
```diff
@@ -17,7 +17,7 @@ module.exports = [{
         }
         return h.view('funding-options/not-valid')
       }
-      return h.view('no-response')
+      return h.redirect('standards?referrer=actions')
```
[the remark this answers is not in the document]
Comment (fareed-mohammed) [reply app/routes/funding-options/actions.js:20]: I didn't change this and in the `main `branch its `return h.redirect('standards?referrer=actions')`

**Review by steve-dickinson — APPROVED** [review]
LGTM

**Pull request merged**